.github/workflows/backend-ci.yml

```
@@ -0,0 +1,62 @@
+    permissions: write-all
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+          token: ${{ secrets.SECRETS_SUBMODULE_ACCESS_TOKEN }}
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
```

actionlint (1.7.7)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)